### PR TITLE
ref: added UWP support and improved demo

### LINKED ISSUE #4: enhance sample app with use cases:
- [x] Register a handled exception.
- [x] Register an unhandled 
- [x] exception.Register user feedback.
- [x] Add Sleep/Resume breadcrumbs
- [x] Nice looking app to simulate a real use-case.

### PR DESCRIPTION
**Pull request by lucas-zimerman**
closes #4.

* Added native integration to UWP.
* Improved Sample app, allowing it to run on iOS, UWP and Android.

The sample app shows the usage of:
* CaptureMessage.
* CaptureException.
* Unhandled Exception Capture.
* User Feedback.
* Automatic Navigation Breadcrumb.
* Native integration with Android, iOS, Android Wear and UWP.
* Additional tags for Android, iOS and UWP.

## Platforms tested:
* Windows 10 with UWP (running .NET Core and .NET Native)
* Android 5.0, Android 10 and Android 9.0 Wear.
* Iphone 8 and iPhone 8 Plus (Simulator only since I lack an iPhone/iPod for testing).

### CODE REVIEW

File: Samples/Sample.Xamarin.Core/App.xaml.cs [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 
-namespace ContribSentry.Sample
+namespace Sample.Xamarin.Core
 {
     public partial class App : Application
     {
         public App()
         {
 
             InitializeComponent();
```
Comment (lucas-zimerman) [inline Samples/Sample.Xamarin.Core/App.xaml.cs:12]: remove extra line

File: Samples/Sample.Xamarin.Core/CustomControls/EditorValidation.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        {
+            var control = (EditorValidation)bindable;
+            control.ErrorMessage = (string)newValue;
+            control.ErrorLabel.Text = control.ErrorMessage;
+        }
+
+        public bool Invalid
+        {
+            get => (bool)GetValue(InvalidProperty);
+            set
```
Comment (lucas-zimerman) [inline Samples/Sample.Xamarin.Core/CustomControls/EditorValidation.xaml.cs:105]: fix format

File: Samples/Sample.Xamarin.Core/CustomControls/EntryValidation.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        {
+            var control = (EntryValidation)bindable;
+            control.ErrorMessage = (string)newValue;
+            control.ErrorLabel.Text = control.ErrorMessage;
+        }
+
+        public bool Invalid
+        {
+            get => (bool)GetValue(InvalidProperty);
+            set
```
Comment (lucas-zimerman) [inline Samples/Sample.Xamarin.Core/CustomControls/EntryValidation.xaml.cs:103]: fix format

File: Samples/Sample.Xamarin.Core/Interfaces/NavigationService.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+using Rg.Plugins.Popup.Services;
+using Sample.Xamarin.Core.Rules;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace Sample.Xamarin.Core.Interfaces
+{
+    /// <summary>
+    /// Hides the UI Tasks
```
Comment (lucas-zimerman) [inline Samples/Sample.Xamarin.Core/Interfaces/NavigationService.cs:11]: fix correct summary

File: Samples/Sample.Xamarin.Core/Resources/PopupSvg.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
```
Comment (lucas-zimerman) [inline Samples/Sample.Xamarin.Core/Resources/PopupSvg.xaml.cs:6]: remove line

File: Samples/Sample.Xamarin.Core/Resources/UnhandledSvg.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
```
Comment (lucas-zimerman) [inline Samples/Sample.Xamarin.Core/Resources/UnhandledSvg.xaml.cs:6]: remove line

File: Samples/Sample.Xamarin.Core/ViewModels/MainPageViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+                DisplayAlert("Whoops", "A handled exception happened", "OK");
+            }
+        };
+
+        private Action Unhandle => () =>
+        {
+            var authService = new AuthService();
+            authService.DoLogin("admin", "1234");
+        };
+
```
Comment (lucas-zimerman) [inline Samples/Sample.Xamarin.Core/ViewModels/MainPageViewModel.cs:81]: remove extra line

File: Samples/Sample.Xamarin.Core/Views/DiscoPage.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+using Xamarin.Forms;
+
+namespace Sample.Xamarin.Core.Views
+{
+    public partial class DiscoPage : ContentPage
+    {
+        private CancellationTokenSource _tokenSource = new CancellationTokenSource();
+        private CancellationToken _token;
+
+
```
Comment (lucas-zimerman) [inline Samples/Sample.Xamarin.Core/Views/DiscoPage.xaml.cs:14]: remove extra line

File: Samples/Sample.Xamarin.Core/Views/LoginPage.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
+﻿using Sample.Xamarin.Core;
+using Sentry.Protocol;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
```
Comment (lucas-zimerman) [inline Samples/Sample.Xamarin.Core/Views/LoginPage.xaml.cs:10]: remove extra line

File: Samples/Sample.Xamarin.uwp/App.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+            this.Suspending += OnSuspending;
+        }
+
+        protected override void OnLaunched(LaunchActivatedEventArgs e)
+        {
+            SentryInitializer.Init();
+
+
+
+
```
Comment (lucas-zimerman) [inline Samples/Sample.Xamarin.uwp/App.xaml.cs:25]: remove extra lines

File: Sentry.Xamarin.Forms/Internals/NativeIntegration.uwp.cs [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+            {
+                var st = exception.StackTrace.ToString();
+                exception.Data[Mechanism.HandledKey] = false;
+                exception.Data[Mechanism.MechanismKey] = "Application.UnhandledException";
+                var @eventId = SentrySdk.CaptureException(exception);
+                SentrySdk.FlushAsync(TimeSpan.FromSeconds(10)).Wait();
+                (_hub as IDisposable)?.Dispose();
+            }
+        }
+
```
Comment (lucas-zimerman) [inline Sentry.Xamarin.Forms/Internals/NativeIntegration.uwp.cs:64]: remove extra line

File: Sentry.Xamarin.Forms/Extensions/PageExtensions.cs [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+            {
+                return new PageInfo()
+                {
+                    Name = page.Title
+                };
+            }
+            return new PageInfo()
+            {
+                Name = type.Name,
+                BaseType = type.BaseType.Name
```
Comment (bruno-garcia) [inline Sentry.Xamarin.Forms/Extensions/PageExtensions.cs:21]: `type.BaseType` can't be null?

File: Sentry.Xamarin.Forms/Internals/NativeIntegration.uwp.cs [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+            _application.LeavingBackground -= OnResume;
+
+            _hub = null;
+        }
+
+        [HandleProcessCorruptedStateExceptions, SecurityCritical]
+        internal void NativeHandle(object sender, Windows.UI.Xaml.UnhandledExceptionEventArgs e)
+        {
+            //We need to backup the reference, because the Exception reference last for one access.
+            //After that, a new  Exception reference is going to be set into e.Exception.
```
Comment (bruno-garcia) [inline Sentry.Xamarin.Forms/Internals/NativeIntegration.uwp.cs:44]: TIL

File: Sentry.Xamarin.Forms/Internals/NativeIntegration.uwp.cs [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+            Unregister();
+            Handle(exception);
+        }
+
+        [HandleProcessCorruptedStateExceptions, SecurityCritical]
+        internal void Handle(Exception exception)
+        {
+            if (exception != null)
+            {
+                var st = exception.StackTrace.ToString();
```
Comment (bruno-garcia) [inline Sentry.Xamarin.Forms/Internals/NativeIntegration.uwp.cs:55]: Some debugging code left over

**Review by bruno-garcia — APPROVED** [review]
Some suggestions but LGTM

**Pull request merged**